File: adapter_pattern/ap.cs
class InternalClass
{
  public string Fetch()
  {
    return "get user info";
  }
}

class ExternalClass
{
  public string Search()
  {
    return "get user data";
  }
}

class Adapter
{
  private ExternalClass _externalClass;

  public Adapter(ExternalClass externalClass)
  {
    _externalClass = externalClass;
  }

  public string Fetch()
  {
    return _externalClass.Search();
  }
}

File: adapter_pattern/ap.cpp
#include <iostream>
#include <string>

// InternalClass
class InternalClass {
public:
  std::string fetch() {
    return "get user info";
  }
};

// ExternalClass
class ExternalClass {
public:
  std::string search() {
    return "get user data";
  }
};

// Adapter
class Adapter : public InternalClass {
private:
  ExternalClass* externalClass;

public:
  Adapter(ExternalClass* externalClass) : externalClass(externalClass) {}

  std::string fetch() override {
    return externalClass->search();
  }
};

int main() {
  ExternalClass external;
  Adapter adapter(&external);

  std::cout << adapter.fetch() << std::endl;

  return 0;
}